documentation/modules/managing/con-consumer-config-properties.adoc

```diff
@@ -267,49 +280,73 @@ If a consumer group or standalone consumer is inactive and commits no offsets du
- 
-Additionally, adjusting the `max.poll.interval.ms` configuration can prevent rebalances caused by prolonged processing tasks, allowing you to specify the maximum interval between polls for new messages.
-Use the `max.poll.records` property to cap the number of records returned from the consumer buffer during each poll. 
-Reducing the number of records allows the consumer to process fewer messages more efficiently.  
-In cases where lengthy message processing is unavoidable, consider offloading such tasks to a pool of worker threads. 
-This parallel processing approach prevents delays and potential rebalances caused by overwhelming the consumer with a large volume of records.
+Poll intervals and record limits:: 
+* Use the `max.poll.interval.ms` property to prevent rebalances caused by prolonged processing tasks by setting the maximum interval between polls.
+* Use the `max.poll.records` property to limit the number of records returned during each poll. 
+Processing fewer messages more efficiently can prevent delays.
 
-[source,shell,subs="+quotes"]
+Session timeout and heartbeat intervals::
+* Use the `session.timeout.ms` property to set a longer timeout to reduce rebalances caused by temporary network glitches or minor processing delays.
+* Adjust the `heartbeat.interval.ms` property to balance failure detection checks with minimizing unnecessary rebalances.
+
+Adopt partition assignment strategies:: Use appropriate partition assignment strategies to reduce the number of partitions that need to be reassigned during a rebalance, minimizing the impact on active consumers.
+
+Monitor consumer health:: Instability in consumer applications, such as frequent crashes, can trigger rebalances. 
+Use Kafka consumer metrics to monitor such things as rebalance rates, session timouts, and failed fetch requests.
+
+.Example configuration to minimize the impact of rebalances
+[source,shell]
 ----
 # ...
-group.instance.id=_UNIQUE-ID_ <1>
-max.poll.interval.ms=300000 <2>
-max.poll.records=500 <3>
+group.instance.id=<unique_id>
+max.poll.interval.ms=300000
+max.poll.records=500
+session.timeout.ms=45000
+heartbeat.interval.ms=3000
+partition.assignment.strategy=org.apache.kafka.clients.consumer.CooperativeStickyAssignor
 # ...
 ----
-<1> The unique instance id ensures that a new consumer instance receives the same assignment of topic partitions.
-<2> Set the interval to check the consumer is continuing to process messages.
-<3> Sets the number of processed records returned from the consumer.
+
+.Scaling strategies
+To minimize the impact of rebalances during scaling of consumer groups, consider the following approaches:
+
+Set a rebalance delay:: Use the `group.initial.rebalance.delay.ms` property in the Kafka configuration to delay the time it takes for consumers to join a new consumer group before performing a rebalance.
```

Comment:
Introducing a delay, avoid triggering several rebalances when you start multiple consumers near the same time. This depends on the orchestration used. I've seen cases where Kubernetes can take ~10 secs to start a dozen consumer pods, in that case it can cause 1 to 2 rebalances.